### DISCUSSION TITLE
permit2_permit decoding issue

### DISCUSSION THREAD

**Discussion by gauravsc** [Elnaril/uniswap-universal-router-decoder#17] [Q&A] [answered]
Great work with this library.
 
I noticed that when decoding a chained permit2_permit in front of a v3_exact_swapt_in, the decoded output shows the `commands` as  `'commands': b'\n\x00'`, when in fact the code for permit2_permit is `0x0a`

**Comment by gauravsc** [comment]
`<Function execute(bytes,bytes[],uint256)>, {'commands': b'\n\x00', 'inputs': [(<Function PERMIT2_PERMIT(((address,uint256,uint256,uint256),address,uint256),bytes)>, {'struct': {'details': {'token': '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2', 'amount': 100000000000000000, 'expiration': 1693576461, 'nonce': 23}, 'spender': '0x3fC91A3afd70395Cd496C647d5a6CC9D4B2b7FAD', 'sigDeadline': 1690984641}, 'data': b"\xf9\xddx\xa7\xc4\xd5Sc\xc7H*f\xdc\xdb\xbb\xcfr\x0fC\xfeM\xcc\xc9\xf1\x05Q0\x7f3\xa7!?]\x040\xc1\xf9\x8e:\xd7\x1a \x94'\x00sG\x0f\x80B\xf9\xbff\xa7\x9d\xd0\xd5\xfa\xbe4\xe2\xf3\xe0A\x1b"}), (<Function V3_SWAP_EXACT_IN(address,uint256,uint256,bytes,bool)>, {'recipient': '0x0000000000000000000000000000000000000001', 'amountIn': 100000000000000000, 'amountOutMin': 205466792480, 'path': b"\xc0*\xaa9\xb2#\xfe\x8d\n\x0e\\O'\xea\xd9\x08<ul\xc2\x00'\x10r\xe4\xf9\xf8\x08\xc4\x9a*a\xde\x9cX\x96)\x89 \xdcN\xee\xa9", 'payerIsSender': True})], 'deadline': 1690982841})`

**Comment by Elnaril** [comment, marked as answer]
yes this is normal:
```
>>> b'\n\x00'.hex()
'0a00'
```

**Reply by gauravsc** [reply]
thanks, I didn't realize that.
